pkg/web/initclient/init_http_handler_test.go

```diff
@@ -0,0 +1,125 @@
+package initclient
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/google/uuid"
+
+	"github.com/aerogear/mobile-security-service/pkg/models"
+
+	"github.com/aerogear/mobile-security-service/pkg/web/apps"
+	"github.com/labstack/echo"
+)
+
+func TestHTTPHandler_InitClientApp(t *testing.T) {
+	type fields struct {
+		appsService apps.Service
+	}
+	type args struct {
+		device models.Device
+	}
+	tests := []struct {
+		name           string
+		fields         fields
+		args           args
+		wantStatusCode int
+	}{
+		{
+			name: "A 400 Bad Request should be returned when request body is missing device ID",
+			args: args{
+				device: models.Device{
```

Comment:
Use the helper to get the mode mocked.

Reply:
I don't see the point. Each case has a different mock device, with different missing properties. We _could_ create a helper function for each of these test cases, but what is the point? We would be writing the same amount of code, except the unit test would now be dependent on an external package.

Reply:
I added [here](https://github.com/aerogear/mobile-security-service/pull/76#discussion_r262887466) the detailed explanation and how I am suggesting you do it. 

The main reasons for you use the mock are:
* Make clear what is diff in each scenario, what changed and what will be checked
* Make easier the understanding and keep maintained. 
* Reduce  significantly the complexity of the func( it is not a good practice have fun with more than X lines which means with high complexity)
* Centralize the implementation which means that if the struct change we need change just the mock instead of understanding all these ones and do change across all tests. For example, we decided to add a property value for the versions then all tests need to be changed + mocks. If you follow the suggestion when you add the new property and value in the mock all will still work well as should be.